test/functional/helpers/session.js

```diff
@@ -18,6 +19,11 @@ async function initSession (caps, remoteOpts = {}) {
   });
   driver.name = undefined;
   driver.errored = false;
+
+  if (!caps.alwaysMatch['appium:usePrebuiltWDA'] && !caps.alwaysMatch['appium:derivedDataPath']) {
```

Comment:
use extractCapabilityValue helper function

Reply:
I rather thought abut something like:

```js
try {
  driver = await remote({
    hostname: HOST,
    port: PORT,
    capabilities: caps,
    connectionRetryTimeout: MOCHA_TIMEOUT,
    connectionRetryCount: 1,
    ...remoteOpts,
   });
  didBuildWda = true;
} catch (e) {
  didBuildWda = false;
  throw e;
}
```

Reply:
Like this? https://github.com/appium/appium-xcuitest-driver/pull/1466/commits/807bf2a9f100c110aa12511150faefaa385e5fcc
Or/and do you prefer to remove these capabilities checks and keep it simple?